### DISCUSSION TITLE
Description retrieval

### DISCUSSION THREAD

**Discussion by Melchior-Blank** [ip-tools/python-epo-ops-client#56] [Q&A] [answered]
Hello and thank you very much for this fantastic library!

I have a question regarding Description Retrieval though. There is an API called Published Data Description Retrieval Service.
For instance, this request: http://ops.epo.org/3.2/rest-services/published-data/publication/docdb/CH715395A2/description
ends up with a full-text description.
I've tried to get the same result by using `client.published_data('publication', epo_ops.models.Docdb('715395', 'CH', 'A2'), constituents=['description']`, but haven't got any description in the response. What am I doing wrong?

Thanks in advance!

Greetings,
Irina

**Comment by Melchior-Blank** [comment, marked as answer]
For whom it might concern, I've just figured out my mistake: 'description' goes to the `endpoint`, not to the list of `constituents`.
This `client.published_data('publication', epo_ops.models.Docdb('715395', 'CH', 'A2'), endpoint = 'description')` works beautifully.

Thanks a lot for your library, it is trully a gift!

Greetings,
Irina

**Reply by gsong** [reply]
@Melchior-Blank Glad you were able to work that out by yourself. I'm glad this package is useful.

**Reply by CholoTook** [reply]
Where are the different endpoints documented?

Perhaps I can guess that they are implied [here](https://developers.epo.org/ops-v3-2/apis), which suggests:
- abstract
- biblio
- claims
- description
- equivalents
- full-cycle
- fulltext
- images

However, in general I'm confused about how to marry the data on that page with the calls in the API...

For example, in the https://github.com/ip-tools/python-epo-ops-client?tab=readme-ov-file#client > method table I see `family`... does that mean that family supports the:
- biblio
- legal

endpoints?

I'm so sorry... this is now turning into my notes live as I explore ;-)

That page has the following H3 elements:
- Classification (COULD BE ADDED?)
- Family (in the client/method table)
- Published (in the client/method table)
- Register (in the client/method table)
- Number (in the client/method table)
- Legal (COULD BE ADDED?)

So perhaps this is what we have:
- Family
  - biblio
  - legal
- Published
  - abstract
  - biblio
  - claims
  - description
  - equivalents
  - full-cycle
  - fulltext
  - images
- Register
  - biblio
  - events
  - procedural-steps
  - upp
- Number
  - err...




Sorry for the above

**Comment by amotl** [comment]
Hi. Thanks for your feedback. We just enhanced the documentation according to your findings. Thanks @Melchior-Blank!

- GH-138

@CholoTook and @xplusyi: Feel free to submit more patches to improve documentation further.
